module/epochs/qc_client.go

```diff
@@ -0,0 +1,133 @@
+package epochs
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+
+	"github.com/onflow/cadence"
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/flow-core-contracts/lib/go/templates"
+
+	sdk "github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/client"
+
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/module"
+)
+
+// QCContractClient is a client to the QC Contract
+type QCContractClient struct {
+	accessAddress   string         // address of the access node
+	contractAddress string         // QuorumCertificate contract address
+	client          *client.Client // flow-go-sdk client to access node
+	me              module.Local   // local node
+}
+
+// NewQCContractClient returns a new client to the QC contract
+func NewQCContractClient(me module.Local, accessAddress string, qcContractAddress string) (*QCContractClient, error) {
+
+	// create a new instance of flow-go-sdk client
+	flowClient, err := client.New(accessAddress, grpc.WithInsecure())
+	if err != nil {
+		return nil, fmt.Errorf("could not create flow client: %w", err)
+	}
+
+	return &QCContractClient{
+		me:              me,
+		accessAddress:   accessAddress,
+		contractAddress: qcContractAddress,
+		client:          flowClient,
+	}, nil
+}
+
+// SubmitVote submits the given vote to the cluster QC aggregator smart
+// contract. This function returns only once the transaction has been
+// processed by the network. An error is returned if the transaction has
+// failed and should be re-submitted.
+func (c *QCContractClient) SubmitVote(ctx context.Context, vote *model.Vote) error {
+
+	// get latest sealed block to execute transaction
+	latestBlock, err := c.client.GetLatestBlock(ctx, true)
+	if err != nil {
+		return fmt.Errorf("could not get latest block from node: %w", err)
+	}
+
+	// attach submit vote transaction template and build transaction
+	// TODO: requires `ENV`?
+	submitVoteScript := templates.GenerateSubmitVoteScript(templates.Environment{})
+	tx := sdk.NewTransaction().
+		SetScript(submitVoteScript).
+		SetGasLimit(1000).
+		SetReferenceBlockID(latestBlock.ID)
+
+	// add signature data to the transaction and submit to node
+	err = tx.AddArgument(cadence.NewString(string(vote.SigData)))
+	if err != nil {
+		return fmt.Errorf("could not add raw vote data to transaction: %w", err)
+	}
+
+	// TODO: what are these?
+	// keyIndex := 0
+	// account := sdk.Account{}
```

Comment:
The key index identifies which key for a given account you are signing with. These will also need to be configuration parameters passed into `QCContractClient`, I think we will need `accountAddress`, `keyIndex`, and `privateKey`. We could then retrieve the key algorithm and hash algorithm needed to construct the `crypto.Signer` object using `client.GetAccount`.

We can make the assumption that the key used for the account the node uses here autonomously has weight 1000.